superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,234 @@
+"""
+
+import os
+import time
+
+# TODO - add mechanism to import torch as needed according to docker
+import torch
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmark
+from superbench.benchmarks.context import Enum
+
+
+class ComputationKernelType(Enum):
+    """The Enum class representing different computation kernel type."""
+    MATMUL = 'matmul'
+    MUL = 'mul'
+
+
+class ComputationCommunicationOverlap(MicroBenchmark):
+    """The base class of micro-benchmarks."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+        # Command lines to launch the micro-benchmarks.
+        self.__commands = list()
+        self.__world_size = 1
+        self.__local_rank = 0
+        torch.backends.cudnn.benchmark = True
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--n',
+            type=int,
+            default=1880,
+            required=False,
+            help='The N dim of matmul (M, K) * (K, N).',
+        )
+        self._parser.add_argument(
+            '--k',
+            type=int,
+            default=1024,
+            required=False,
+            help='The K dim of matmul (M, K) * (K, N) or The K dim or mul (M, K)*(M, K). ',
+        )
+        self._parser.add_argument(
+            '--m',
+            type=int,
+            default=640,
+            required=False,
+            help='The M dim of matmul (M, K) * (K, N)  or The m dim or mul (M, K)*(M, K). ',
+        )
+        self._parser.add_argument(
+            '--p',
+            type=int,
+            default=5000,
+            required=False,
+            help='The P dim of Tensor(P, Q) to transfer in Nccl AllReduce.',
+        )
+        self._parser.add_argument(
+            '--q',
+            type=int,
+            default=1400,
+            required=False,
+            help='The Q dim of Tensor(P, Q) to transfer in Nccl AllReduce.',
+        )
+        self._parser.add_argument(
+            "--ratio",
+            type=int,
+            default=30,
+            required=False,
+            help="The execution ratio number between computation kernel and nccl kernel",
+        )
+        self._parser.add_argument(
+            '--kernel',
+            type=ComputationKernelType,
+            default=[ComputationKernelType.MUL],
+            required=False,
+            help='Computation kernel type. E.g. {}.'.format(' '.join(ComputationKernelType.get_values())),
```

Comment:
accept one kernel or a list of kernels?

Reply:
To merge matmul and mat as one benchmark, here should be list of kernels.